ethereum/ethevents/events.go

```diff
@@ -377,20 +400,28 @@ func (ep *EventProcessor) next() *ethtypes.Log {
 
 func (ev *EthereumEvents) runEventProcessor(ctx context.Context) {
 	ev.EventProcessor.eventProcessorRunning = true
+	log.Info("starting event processor routine")
 	go func() {
 		for {
-			event := <-ev.EventProcessor.Events
-			evtJSON, err := event.MarshalJSON()
-			if err != nil {
-				log.Error(err)
-				continue
-			}
-			if event.Removed {
-				log.Warnf("removing reversed log event: %s", evtJSON)
-				ev.EventProcessor.del(&event)
-			} else {
-				log.Debugf("queued event log: %s", evtJSON)
-				ev.EventProcessor.add(&event)
+			select {
+			case <-ctx.Done():
+				return
+			case event := <-ev.EventProcessor.Events:
+				evtJSON, err := event.MarshalJSON()
+				if err != nil {
+					log.Error(err)
+					continue
+				}
+				if event.Removed {
+					log.Warnf("removing reversed log event: %s", evtJSON)
+					ev.EventProcessor.del(&event)
+				} else {
+					if event.BlockNumber == 0 {
+						log.Fatalf("this event %+v must not be arribe here (BOMB)", event)
```

Comment:
what's the meaning of this message? I don't get it. Can you rephrase?

Reply:
just a sanity check cause I do not expect having block number 0 on event logs. Changed the message for more clarity